p2p/discovery.py

```diff
@@ -680,13 +709,63 @@ def send_find_node_v5(self, node: kademlia.Node, target_node_id: int) -> None:
         message = _pack_v5(CMD_FIND_NODE.id, (node_id, _get_msg_expiration()), self.privkey)
         self.send_v5(node, message)
 
-    def send_topic_query(self, node: kademlia.Node, topic: Hash32) -> None:
-        self.logger.trace('>>> topic_query to %s', node)
+    def send_topic_query(self, node: kademlia.Node, topic: bytes) -> Hash32:
+        self.logger.trace('>>> topic_query (%s) to %s', topic, node)
         payload = (topic, _get_msg_expiration())
         message = _pack_v5(CMD_TOPIC_QUERY.id, payload, self.privkey)
-        # TODO: Get the msg hash and store it in the ticket store to match against received
-        # topic_node msgs.
-        self.send_v5(node, message)
+        return self.send_v5(node, message)
+
+    def send_topic_nodes(
+            self, node: kademlia.Node, echo: Hash32, nodes: Tuple[kademlia.Node, ...]) -> None:
+        encoded_nodes = []
+        for n in nodes:
+            encoded_nodes.append(n.address.to_endpoint() + [n.pubkey.to_bytes()])
```

Comment:
Any reason not to use a comprehension to do this.  I think it's more efficient and maybe reads a bit easier.

```python
encoded_nodes = tuple(
    n.address.to_endpoint() + [n.pubkey.to_bytes()]
    for n in nodes
)